### DISCUSSION TITLE
Steam Next Fest June 2022 - Demo AppIds

### DISCUSSION THREAD

**Discussion by TCNOco** [TCNOco/TcNo-Mass-Steam-Idler#1] [Announcements]
The [Steam Next Fest 2022](https://store.steampowered.com) is now live, and you can level up a badge for a total of 100XP by simply downloading and launching demos.

This software lets you launch each app for a second, registering you playing it and poof, another level is added.

## Download App ID text files
**List of 1,201 App IDs (includes everything)**: Taken from [xPaw's](https://steamcommunity.com/id/xPaw/games/) recently played: Download the [appids.txt here](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8918897/appids.txt) - A user with badge level 1,049 (appears to be the max level).

**List of 986 sale-specific entries (Using method below, Updated 2022-06-19):** [appids.txt](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8935585/appids.txt)

**List of 57 App IDs**: Check [appids.txt](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8910539/appids.txt), and add it to the install folder to run these. [This is the updated list from the comment below].

When you've finished idling everything, create `skipcheck.txt` in the install folder, and delete `idle_queue.txt`. Run the program again, and it will idle through EVERYTHING again, making sure that every game was idled. This takes a long time but can get you a few levels higher usually.

## PLEASE NOTE
You can only activate 50 free games/demos per hour. The program does its best to explain this. If the Steam Store starts opening or you receive SteamAPI errors in the console, wait an hour and retry.

## Done? Remove demos easily!
To remove demos from your account quickly, check the guide [HERE](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/blob/main/Mass_Remove.md)

### Previous lists:
List of 832 App IDs: Also, this list from [Reddit](https://www.reddit.com/r/FreeGameFindings/comments/vbhniu/comment/icfkwfh/?utm_source=share&utm_medium=web2x&context=3). Download the [appids.txt here](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8914561/appids.txt).

**Comment by TCNOco** [comment]
Update: 
[NextFestJune2022-AppIDs.txt](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8910203/NextFestJune2022-AppIDs.txt) is all the Steam game IDs I found on the sale. 1,181 AppIds.
Getting the demo IDs for each: 
[NextFestJune2022-DemoAppIds.txt](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8910217/NextFestJune2022-DemoAppIds.txt) are what I got back. 983 AppIds (duplicates removed).

Most of these, when run as `steam://install/<appid>`, take you to the store page instead of activating and prompting for game download. Odd. Even the download demo button on the page refreshes instead of popping up with a download. I have no idea why this is.

 After trying to activate everything and then checking which ones are activated on a second run, I'm left with these:
[NextFestJune2022-ValidDemoAppIds.txt](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8910237/NextFestJune2022-ValidDemoAppIds.txt). 61 demo app IDs that seem to activate reliably.

**Reply by omarovi55** [reply]
Absolutely working perfectly. Thank you so much for sharing such a beautiful trick

**Comment by PatrickJnr** [comment]
Any new AppIDs to use? since I'm sure there are more to farm, judging by this list https://steamdb.info/badge/60

1044 as of this post

**Reply by TCNOco** [reply]
## My method

Opening the browser console and copy/pasting jQuery (https://code.jquery.com/jquery-3.6.0.min.js), then running this:
```js
var allAppIds = [];
var countTo = 0;
var currentOffset = 0;
function uniq_fast(a) {var seen={};var out=[];var len=a.length;var j=0;for(var i = 0; i < len; i++) {var item = a[i];if(seen[item] !== 1) {seen[item] = 1;out[j++] = item;}} return out;
}
// Run the above only once.
// Re-run below until complete.
$.getJSON(`https://store.steampowered.com/saleaction/ajaxgetsaledynamicappquery?cc=ZA&l=english&&flavor=topwishlisted&strFacetFilter=&tabuniqueid=13&return_capsules=true&origin=https:%2F%2Fstore.steampowered.com&bForceUseSaleTag=true&strTabFilter=&bRequestFacetCounts=true&clanAccountID=39049601&clanAnnouncementGID=3337742851854054341&count=100&start=${currentOffset}`, function(data) {
    allAppIds = uniq_fast(allAppIds.concat(data["appids"]));
    countTo = data["match_count"];
   
    console.log(allAppIds);
    console.log("Count to: " + countTo + ", current: " + currentOffset);
    currentOffset += data["appids"].length;
});
```
[Re-running as needed]
This is what I get: [MyAppIds.txt](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8914405/MyAppIds.txt)


Converting these AppIDs to Demo AppIDs:

 Adding `var allAppIds = ` to the start of the above, I used:
```js
var allAppIds = [<<aforementioned list>>];
var allAppIdsDemo = [];
current = 0;
function getDemo(sid){
  return $.getJSON("https://store.steampowered.com/saleaction/ajaxgetdemoevents?appids[]="+sid).then(function(data){
    return data["info"][0]["demo_appid"];
  });
}

function proc(s) {
	getDemo(allAppIds[current]).then(function(returndata){
	  current++;
	  allAppIdsDemo.push(allAppIds[current] + " - " + returndata);
	  if (allAppIdsDemo.length < allAppIds.length){
		proc(allAppIds[current]);
	  }
	  console.log(allAppIdsDemo.length + " of " + allAppIds.length);
	});
}

proc(allAppIds[0]);
```

This then loops, and I can eventually run `copy(allAppIdsDemo)` to get a list of demo AppIds... Most of which seem to not work :/ -- To get just the demo IDs use `var out = []; allAppIdsDemo.forEach((e) => {out += e.split(" - ")[1] + ",";}); console.log(out);copy(out);`
I have found a list on Reddit that I've edited and included in the original post now.


With the link you posted, there MUST be a much better way of doing this...

**Reply by TCNOco** [reply]
Checking one of the highest users on that leaderboard: https://steamcommunity.com/id/revadike/games?tab=recent

```js
var list = [];
$(".gameListRow").each((i, e) => {
  if ($(e).text().indexOf("Demo") != -1){
    console.log($(e).text().indexOf("Demo"));
    var parts = $(e).find("a").attr("href").split('/');
    list.push(parts[parts.length - 1]);
  }
});
console.log(list);
```
This is the list that was returned: [HUGEDemoList.txt](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8914651/HUGEDemoList.txt). 2,239 entries. That's a LOT. Well, time to run this overnight lol.

Though unsurprisingly, these almost all open my Steam Store instead of registering on my account. Idk what causes this...

I can't remember but I saw some link that can activate things. Some `.../registergame` or `.../activategame`, or something like that... Maybe that can help? I just can't find it anywhere.

**Reply by PatrickJnr** [reply]
Amazing writeup! I wonder why it opens steam :c

**Reply by PatrickJnr** [reply]
You could try **app_license_request** that's a steam console command. could work, may not.

**Reply by ham1tcagdas** [reply]
i tried it for 1 hour repeatedly did not works for me. anyway i m tired. goodluck, reply my comment, if it works really.

**Reply by TCNOco** [reply]
THAT'S WHAT IT IS! I googled "app_license_request", and found this [post](https://www.steamgifts.com/discussion/7ax3D/free-2335-free-steam-things-mega-list-v22)
```
REMEMBER! max 50 appids per hour.
```
THIS is what's wrong. This is the error! This is why it opens the store page!

I ran the idler in the morning, and low and behold: Number 51 takes me to the store page!

![image](https://user-images.githubusercontent.com/10319195/174063748-c48e2182-05c0-4ad3-99a5-ca86fd0797d0.png)


Time to update the idler with info.

Now to wait an hour, test the activation command instead, and see if it's easier for users to copy/paste. It should be faster. Maybe there's a CLI command for it?

**Reply by TCNOco** [reply]
Alright, I've pushed an update that lets the user know of the 50-game activation limit per hour. This should fix the issue everyone, including me, is having.

That being said, I'll look into the app_license_request as soon as my hour passes. It May be useful for some users to copy/paste to activate everything (Just need to verify that all DO activate), instead of waiting 5 seconds per game. This needs to be communicated well, and give the user the hands-off and hands-on method. Opening the console programmatically is easy enough.

**Reply by ham1tcagdas** [reply]
it checks for 1 account only when i try my second steam acc it wont check first checked games again. it wants more new ids

**Reply by TCNOco** [reply]
> it checks for 1 account only when i try my second steam acc it wont check first checked games again. it wants more new ids

Either make a new folder or delete all the text files. This is a good point I'll extend it to support multiple accounts soon.

**Reply by TCNOco** [reply]
Alright, I have a pre-release out now that I currently don't have the time to test. Should let you copy/paste commands to activate 50 at a time https://github.com/TcNobo/TcNo-Mass-Steam-Idler/releases/tag/2022-06-16_01

**Reply by PatrickJnr** [reply]
Aye! glad to have helped in some part :D it seems 1052 is the new max now.

**Reply by Poetic-Abortionist** [reply]
Here's [11K demo ids](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8928823/appids11K.txt) that might or might not work.

**Reply by Akzyra** [reply]
If you already "played" the following IDs the max is 1053, if not you can only get 1039

**removed from next fest (12)** \
2004350,1994580,1991560,1964710,1952830,1946000,
1890150,1788550,1666580,1666350,1595680,1345460,

**demo completely removed (2)** \
2008160,1870980

**Reply by TCNOco** [reply]
Update for 2022-06-19: Running my script pulled game IDs: [2022-06-19.txt](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8935190/2022-06-19.txt) - 986. Getting demo AppIDs: [appids.txt](https://github.com/TcNobo/TcNo-Mass-Steam-Idler/files/8935221/appids.txt) - 986 entries

**Reply by SekiBetu** [reply]
> If you already "played" the following IDs the max is 1053, if not you can only get 1039
> 
> **removed from next fest (12)** 2004350,1994580,1991560,1964710,1952830,1946000, 1890150,1788550,1666580,1666350,1595680,1345460,
> 
> **demo completely removed (2)** 2008160,1870980

You mean that some demos were removed during this festival?

**Reply by TCNOco** [reply]
Yes, a lot actually.

**Reply by SekiBetu** [reply]
> Yes, a lot actually.

so these games won't show in the recent played? i used xPaw's recently played 1201 games list, i only leveled up my badge to 1028

**Reply by TCNOco** [reply]
No, they do show and stay on your account. Apps/Games/Demos removed from Steam still stay on accounts they were activated on before removal.

**Comment by davidewNW** [comment]


**Reply by ham1tcagdas** [reply]
it doesnt wait for 50 minutes. can u make it dos screen wait 50 minutes really.

**Reply by TCNOco** [reply]
It already does this. Run the one ending in Auto. It activates, idles the games and then activates more, etc.

**Reply by ham1tcagdas** [reply]
After 50 completed it starts waiting and adding almost 300 in failed text file under 1 hour and it passes these 300 i think

**Comment by jakey1995abc** [comment]
I've been using SteamDB's free package script almost daily for the last 5+ years, I have every free package available to date claimed on my account. Yet after running TcNo-Mass-Steam-Idler multiple times with the recommended appid list from xpaw's account (and also trying my own custom list with all 10,000+ appids linked to my account) my badge level has only gone up to 626 and stopped counting. Am I doing something wrong or has something changed with how it counts the levels for the badge?

**Reply by TCNOco** [reply]
No idea. Mine stopped going up, but after a short while it went up again. This project was more a proof of concept than anything, a way to idle a lot of games for X time and will be improved, of course. No idea the quirks behind Steam game activation and idling to be honest. Mine is counting up again, however, using today's list (986 entries)

**Reply by BakerIsCoding** [reply]
It's possible to use this like SteamDB's free package script?

**Reply by jakey1995abc** [reply]
> It's possible to use this like SteamDB's free package script?

From what I can see, yes. Although I haven't tried it myself since I use the SteamDB script daily

**Reply by TCNOco** [reply]
You can use that as normal to activate keys - This isn't anything different, other than "opening the game" for X seconds. Not sure why you'd need that though.
